### DISCUSSION TITLE
Create Tandoor Menu

### DISCUSSION THREAD

**Discussion by smilerz** [TandoorRecipes/recipes#2642] [General]
Still very much in BETA - but I have created a python application that will [generate a menu](https://github.com/smilerz/tandoor-menu-generator/tree/master) based off of your Tandoor recipes.

It uses a linear programming algorithm to choose recipes from a robust set of criteria.
For example, you can ask it to:
- Choose 22 recipes
- 7 from keyword breakfast
- 7 from keyword lunch
- 7 from keyword dinner
- 1 from dessert 
- 2 or more should be vegetarian
- 6 must come from the Italian recipe book, but only if not cooked in the last 30 days
- 6 should include a protein (but no liver)
- 10 should be rated at least 4 stars 
- 0 should be rated 2 or less
- 5 should be something I've never cooked
- 2 should be something that I added to tandoor in the last 60 days

Over the next several weeks I hope to add automatically adding items to a meal plan and injecting recipes into SVG templates.

**Comment by vabene1111** [comment]
that sounds very interesting, I was thinking about menus every now and then over the years, maybe we could add a real "Menu" model that holds the menu and some attributes of it.
